modules/spring-config/src/main/resources/META-INF/mule-module.xsd

```diff
@@ -150,24 +150,42 @@
                         </xsd:documentation>
                     </xsd:annotation>
                 </xsd:attribute>
-                <xsd:attribute name="namespace" use="required">
+                <xsd:attribute name="category" type="categoryType" default="COMMUNITY">
                     <xsd:annotation>
                         <xsd:documentation>
-                            Expected namespace of the module to look for when generating the schemas.
+                            Expected category of the module.
                         </xsd:documentation>
                     </xsd:annotation>
                 </xsd:attribute>
-                <xsd:attribute name="category" type="categoryType" default="COMMUNITY">
+                <xsd:attribute name="vendor" type="xsd:string" default="MuleSoft">
                     <xsd:annotation>
                         <xsd:documentation>
-                            Expected category of the module.
+                            Expected vendor of the module.
                         </xsd:documentation>
                     </xsd:annotation>
                 </xsd:attribute>
-                <xsd:attribute name="vendor" type="xsd:string" default="MuleSoft">
+                <xsd:attribute name="minMuleVersion" type="xsd:string" use="optional" default="4.0.0">
                     <xsd:annotation>
                         <xsd:documentation>
-                            Expected vendor of the module.
+                            Min Mule version that the extension requires to work correctly.
+                        </xsd:documentation>
+                    </xsd:annotation>
+                </xsd:attribute>
+                <!-- DSL XML attributes -->
+                <xsd:attribute name="namespace" use="optional">
+                    <xsd:annotation>
+                        <xsd:documentation>
+                            Expected namespace of the module to look for when generating the schemas. If left empty it will
+                            default to http://www.mulesoft.org/schema/mule/[prefix], where [prefix] is the attribute prefix attribute value.
+                        </xsd:documentation>
+                    </xsd:annotation>
+                </xsd:attribute>
+                <xsd:attribute name="prefix" use="optional">
```

Comment:
what the @Xml has is two attributes: `namespace` and `namespaceLocation` that maps to the prefix and namespace.
To my understanding, there's no `namespaceLocation` and it's confusing as it's being used as the namespace of an extension. From wikipedia: `...A namespace name is a uniform resource identifier (URI). ...`, which means having something like [`mulesoft`](https://github.com/mulesoft/mule-extensions-api/blob/master/mule-extensions-api/src/main/java/org/mule/runtime/extension/api/annotation/dsl/xml/Xml.java#L32) does not comply to a URI, but the [`schemaLocation`](https://github.com/mulesoft/mule-extensions-api/blob/master/mule-extensions-api/src/main/java/org/mule/runtime/extension/api/annotation/dsl/xml/Xml.java#L41) does.